### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Vasyl-Poremchuk**

### CODE REVIEW

**Review by mltdll — APPROVED** [review]

File: services/movie_session.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    if cinema_hall_id:
+        updated_movie_session.cinema_hall_id = cinema_hall_id
+
+    updated_movie_session.save()
+
+    return updated_movie_session
+
+
+def delete_movie_session_by_id(session_id: int) -> MovieSession:
```
Comment (Nattalli) [inline services/movie_session.py:47]: does it really return `MovieSession` object?

File: db/models.py [hunk 2]
```diff
@@ -14,3 +14,38 @@ class Actor(models.Model):
 
     def __str__(self) -> str:
         return f"{self.first_name} {self.last_name}"
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    actors = models.ManyToManyField(Actor)
+    genres = models.ManyToManyField(Genre)
```
Comment (Nattalli) [inline db/models.py:23]: add `related_name`

File: services/cinema_hall.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from db.models import CinemaHall
+
+
+def get_cinema_halls() -> CinemaHall:
```
Comment (Nattalli) [inline services/cinema_hall.py:4]: returns `Queryset`

File: services/movie.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+def get_movies(
+        genres_ids: list[int] = None, actors_ids: list[int] = None
+) -> Movie:
+    queryset = Movie.objects.all()
+
+    if genres_ids and actors_ids:
+        queryset = queryset.filter(
+            genres__id__in=genres_ids, actors__id__in=actors_ids
+        )
```
Comment (Nattalli) [inline services/movie.py:12]: btw, it is unnecessary lines as your next code will do the same

**Review by Nattalli — CHANGES_REQUESTED** [review]

**Review by Nattalli — APPROVED** [review]
GJ!

**Pull request closed without merging**